bibblio.py

```diff
@@ -0,0 +1,377 @@
+import json
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from nose.tools import set_trace
+from urlparse import urlparse
+
+from bs4 import BeautifulSoup
+from sqlalchemy.orm import eagerload
+
+from core.coverage import (
+    CoverageFailure,
+    CoverageProvider,
+    CoverageRecord,
+)
+from core.model import (
+    Credential,
+    CustomList,
+    CustomListEntry,
+    DataSource,
+    Edition,
+    Hyperlink,
+    Identifier,
+    Representation,
+    Resource,
+    Work,
+)
+from core.util.epub import EpubAccessor
+from core.util.http import HTTP
+
+from config import Configuration
+
+
+class BibblioAPI(object):
+
+    API_ENDPOINT = u'https://api.bibblio.org/v1/'
+    CATALOGUES_ENDPOINT = API_ENDPOINT + u'catalogues/'
+    CONTENT_ITEMS_ENDPOINT = API_ENDPOINT + u'content-items/'
+
+    TOKEN_CONTENT_TYPE = u'application/x-www-form-urlencoded'
+
+    log = logging.getLogger(__name__)
+
+    @classmethod
+    def from_config(cls, _db):
+        config = Configuration.integration(Configuration.BIBBLIO_INTEGRATION)
+        if not (config and len(config.values())==2):
+            return None
+
+        client_id = config.get(Configuration.BIBBLIO_ID)
+        client_secret = config.get(Configuration.BIBBLIO_SECRET)
+        return cls(_db, client_id, client_secret)
```

Comment:
Here are the "external integrations" (across all our applications) I could find that aren't a collection or an authentication technique:

* Metadata wrangler
* Google Analytics
* New York Times API
* Amazon S3
* NYPL Shadowcat
* Content Cafe
* The CDN
* Loggly
* Elasticsearch (???)

Basically everything that lives in the 'integrations' section of the JSON. A lot of these are random enough that I think it makes sense to just give them a named `ExternalIntegration`. But that can happen in a separate core branch.